### PR TITLE
feat(experiments): experiments compare table

### LINKED ISSUE #3472: [datasets][ui] experiment comparison page
Show the detail of a specific experiment

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #3472 

Adds a table to compare experiments.

Screenshot 2024-06-17 at 9 56 39 AM

### CODE REVIEW

File: app/src/pages/embedding/PointSelectionTable.tsx [hunk 1]
```diff
@@ -186,6 +186,7 @@ export function PointSelectionTable({
     state: {
       sorting,
     },
+    columnResizeMode: "onChange",
```
Comment (axiomofjoy) [inline app/src/pages/embedding/PointSelectionTable.tsx:189]: Is this change included here on purpose? Seems unrelated to the PR.

File: app/src/pages/experiment/ExperimentComparePage.tsx [hunk 2]
```diff
@@ -24,15 +37,31 @@ export function ExperimentComparePage() {
...
+        <Suspense fallback={<Loading />}>
+          <ExperimentCompareTable
+            datasetId={data.dataset.id}
+            experimentIds={experimentIds}
+          />
+        </Suspense>
+      ) : (
+        <View padding="size-200">
+          <Alert variant="info" title="No Experiment Selected">
+            Please select at least 1 experiment
```
Comment (axiomofjoy) [inline app/src/pages/experiment/ExperimentComparePage.tsx:61]:
```suggestion
            Please select one or more experiments.
```

Nit

File: app/src/components/table/CompactJSONCell.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import { JSONText } from "@phoenix/components/code/JSONText";
+
+const MAX_LENGTH = 100;
+
+/**
+ * A table cell that is designed to show JSON in a compact form.
+ * It will truncate the text if it is too long.
+ */
+
```
Comment (axiomofjoy) [inline app/src/components/table/CompactJSONCell.tsx:12]:
```suggestion
```

File: app/src/components/code/JSONText.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    const singleValue = obj[key];
+    if (typeof singleValue === "string") {
+      const singleValueStr: string = hasMaxLength
+        ? formatText(singleValue, maxLength)
+        : singleValue;
+      return <span title={fullValue}>{singleValueStr}</span>;
+    }
+  }
+  const textValue = hasMaxLength ? formatText(fullValue, maxLength) : fullValue;
+  return <span>{textValue}</span>;
```
Comment (axiomofjoy) [inline app/src/components/code/JSONText.tsx:45]: Add title here as well?

```suggestion
  return <span title={fullValue}>{textValue}</span>;
```

**Review by axiomofjoy — APPROVED** [review]

File: app/src/pages/embedding/PointSelectionTable.tsx (hunk 1 above)
Comment (mikeldking) [reply app/src/pages/embedding/PointSelectionTable.tsx:189]: you are right, I must have edited the wrong file

Comment (mikeldking) [inline app/src/pages/embedding/PointSelectionTable.tsx:189]:
```suggestion
```
